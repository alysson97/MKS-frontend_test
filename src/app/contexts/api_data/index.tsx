"use client";

import { createContext, useState, useEffect } from "react";

type Produto = {
  id: number;
  name: string;
  photo: string;
  description: string;
  price: number;
  // Adicione outras propriedades conforme necessÃ¡rio
};

type AppContextType = {
  produtos: Produto[];
  carrinho: Produto[];
  adicionarAoCarrinho: (produto: Produto) => void;
  removerDoCarrinho: (produto: Produto) => void;
};

type AppProps = {
  children: React.ReactNode;
};

export const AppContext = createContext<AppContextType>({
  produtos: [],
  carrinho: [],
  adicionarAoCarrinho: () => {},
  removerDoCarrinho: () => {},
});



const ApiDataProvider:  React.FC<AppProps> = ({children})  => {
  const [produtos, setProdutos] = useState<Produto[]>([]);
  const [carrinho, setCarrinho] = useState<Produto[]>([]);

  useEffect(() => {
    const fetchProdutos = async () => {
      const response = await  fetch("https://mks-frontend-challenge-04811e8151e6.herokuapp.com/api/v1/products?page=1&rows=15&sortBy=id&orderBy=ASC", { cache: "no-store" });
      const data = await response.json();
      setProdutos(data.products);
    };
    fetchProdutos();
  }, []);

  const adicionarAoCarrinho = (produto: Produto) => {
    if(!carrinho.find(item=>item.id == produto.id))
      setCarrinho([...carrinho, produto]);
  };

  const removerDoCarrinho = (produto: Produto) => {
    setCarrinho(carrinho.filter(item => item.id !== produto.id));
  };

  return (
    <AppContext.Provider value={{ produtos, carrinho, adicionarAoCarrinho, removerDoCarrinho }}>
      {children}
    </AppContext.Provider>
  );
};

export default ApiDataProvider;
