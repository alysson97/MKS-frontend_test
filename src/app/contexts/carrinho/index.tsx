'use client'
 
import { createContext, useState } from 'react';


type CarrinhoContextType = {
  carrinhoAberto: boolean;
  setCarrinhoAberto: (aberto: boolean) => void;
};
 
export const CarrinhoContext = createContext<CarrinhoContextType>({
  carrinhoAberto: true,
  setCarrinhoAberto: () => {}
});
 
export default function CarrinhoStateProvider({
  children,
}: {
  children: React.ReactNode
}) {

  const [carrinhoAberto, setCarrinhoAberto] = useState(true);

  const [ itemCount, setCount ] = useState(0);


  return <CarrinhoContext.Provider value={{carrinhoAberto, setCarrinhoAberto }}>{children}</CarrinhoContext.Provider>
}