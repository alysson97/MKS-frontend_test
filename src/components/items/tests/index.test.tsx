import { render } from '@testing-library/react';
import Page from '..';
import { AppContext } from "@/app/contexts/api_data";

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: jest.fn(() => ({
    produtos: [{ id: 1, name: 'Iphone 11 128 GB', price: 5000.00, photo: 'https://mks-sistemas.nyc3.digitaloceanspaces.com/products/iphone11x128.webp', description: 'Grave vídeos 4K, faça belos retratos e capture paisagens inteiras com o novo sistema de câmera dupla.' }],
    carrinho: [],
    adicionarAoCarrinho: jest.fn(),
    removerDoCarrinho: jest.fn()
  }))
}));

describe('Page Component', () => {
  it('renders correctly', () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });
});