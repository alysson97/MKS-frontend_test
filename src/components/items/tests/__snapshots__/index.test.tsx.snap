// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Component renders correctly 1`] = `
<div>
  <div
    class="produtos-container flex justify-center"
  >
    <div
      class="produtos grid place-content-center lg:grid-cols-4 md:grid-cols-2"
    >
      <div
        class="card m-2 rounded-xl shadow-lg flex flex-col justify-between"
      >
        <img
          class="cardImage"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/iphone11x128.webp"
        />
        <div
          class="cardText px-2 "
        >
          <div
            class="itemPrice flex justify-between"
          >
            <p
              class="itemName pr-1"
            >
              Iphone 11 128 GB
            </p>
            <p
              class="price p-1 rounded"
            >
              R$
              5000
            </p>
          </div>
          <p
            class="description my-2"
          >
            Grave vídeos 4K, faça belos retratos e capture paisagens inteiras com o novo sistema de câmera dupla.
          </p>
        </div>
        <button
          class="comprarButton py-1 rounded flex justify-center self-end"
        >
          <img
            alt=""
            class="pr-1 mt-1"
            src="/img/comprar.png"
          />
          Comprar
        </button>
      </div>
    </div>
  </div>
</div>
`;
